### PR TITLE
[ts-migration] support prim::SetAttr and fix prim::GetAttr

### PR DESCRIPTION
**Pull request by BoyuanFeng**
- Lifting Tensor Constant attributes to buffers: TorchScript does not automatically lift tensor constant attributes to buffers. So previous converter cannot access tensor constant attributes. This PR fixed the issue.
- Add SetAttr support for tensor attributes by copy_.
- Add SetAttr support for non-tensor attributes. In particular, we maintain the current value of non-tensor attributes in `name_to_non_tensor_attribute_node`, similar to an interpreter pass on non-tensor attributes. So we can support the following use case:
```python
 def forward(self, x):
      c1 = self.count
      self.count += 1
      c2 = self.count
      return x + c1 + c2
```
- Fixed a bug in GetAttr to support the following use case:
```python
def forward(self, inp):
  x = self.buffer
  self.buffer += 1
  y = self.buffer
  return x + y + inp
```

### CODE REVIEW

File: test/export/test_converter.py [hunk 1]
```diff
@@ -17,45 +17,62 @@
...
-        for ts_model in model_list:
-            ep = TS2EPConverter(ts_model, inp).convert()
-            ep_list.append(ep)
-            ep_out, _ = pytree.tree_flatten(ep.module()(*inp))
-            orig_out, _ = pytree.tree_flatten(ts_model(*inp))
-
-            # Check module.
-            if isinstance(mod, torch.nn.Module):
-                self.assertEqual(
-                    ep.state_dict.keys(),
```
Comment (jiashenC) [inline test/export/test_converter.py:44]: nit: We should probably keep those to check the `state_dict` are still equal?

File: test/export/test_converter.py [hunk 2]
```diff
@@ -17,45 +17,62 @@
...
-            # Check results.
-            self.assertEqual(len(ep_out), len(orig_out))
-            for ep_t, orig_t in zip(ep_out, orig_out):
-                if isinstance(ep_t, torch.Tensor) and isinstance(orig_t, torch.Tensor):
-                    self.assertEqual(ep_t.shape, orig_t.shape)
-                    self.assertTrue(torch.allclose(ep_t, orig_t))
-                else:
-                    self.assertEqual(type(ep_t), type(orig_t))
-                    self.assertEqual(ep_t, orig_t)
+                ep = TS2EPConverter(ts_model, inp).convert()
```
Comment (jiashenC) [inline test/export/test_converter.py:58]: Just for my knowledge, why do we need to to convert it for number of iterations?

If it is for checking persistence, should we convert it once and run the module for number of iterations?

File: torch/_export/converter.py [hunk 3]
```diff
@@ -773,3 +802,36 @@ def retrace_as_exported_program(
...
+                root_attr_name = attribute_map[input_name]
+                attr_fqn = (
+                    f"{root_attr_name}.{attr_name}" if root_attr_name else attr_name
+                )
+                value = get_attr(attr_fqn)
+
+                attribute_map[output_name] = attr_fqn
+                if isinstance(value, torch.Tensor):
+                    # Lift tensor constants to be a buffer
+                    self.name_to_buffer_map[attr_fqn] = value
```
Comment (angelayi) [inline torch/_export/converter.py:837]: Should we only lift it if there's a setattr? If it's only a getattr could we just keep it as a tensor constant?

File: torch/_export/converter.py [hunk 4]
```diff
@@ -735,17 +761,20 @@ def __init__(
             else dict()
         )
 
+        self.lift_tensor_constants_to_buffer()
```
Comment (angelayi) [inline torch/_export/converter.py:764]: I would prefer to put this in the convert() function

**Review by angelayi — COMMENTED** [review]
It might be too much work to do in this PR, but it would be good to also test what happens if there is a setattr within the prim::If

File: torch/_export/converter.py (hunk 4 above)
Comment (angelayi) [reply torch/_export/converter.py:827]: ohhh ok

**Review by angelayi — COMMENTED** [review]
awesome! please add a comment to the `lift_tensor_constants_to_buffer` function so that we know why this function exists (we want to lift the tensor constant attributes as buffers).

also if you have time can you update the PR summary to quickly describe what you're doing: (1) lifting tensor constant attributes to buffers, (2) adding setattr support for tensor attributes through inserting a copy_, and then having retracing handle the mutations, (3) adding setattr support to non-tensor attributes, which should get inlined when retracing

**Review by angelayi — APPROVED** [review]

**Pull request closed without merging**